terragrunt/modules/ecs-app/deployment.tf

```diff
@@ -0,0 +1,227 @@
+// permissions needed while creating the task.
+
+resource "aws_iam_role" "task_execution" {
+  name = "ecs-task-execution--${var.name}"
+  assume_role_policy = jsonencode({
+    Version = "2008-10-17"
+    Statement = [
+      {
+        Sid    = "ECS"
+        Effect = "Allow"
+        Action = "sts:AssumeRole"
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "task_execution" {
+  role = aws_iam_role.task_execution.name
+  name = "permissions"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      // Access to SSM Parameter Store
+      {
+        Sid    = "AllowParameterStore"
+        Effect = "Allow"
+        Action = "ssm:GetParameters"
+
+        Resource = concat(
+          values(data.aws_ssm_parameter.task).*.arn,
+          values(var.computed_secrets),
+        )
+      },
+
+      // Access to CloudWatch Logs
+      {
+        Sid    = "AllowLogs"
+        Effect = "Allow"
+        Action = [
+          "logs:PutLogEvents",
+          "logs:CreateLogStream",
+        ]
+        Resource = "${aws_cloudwatch_log_group.task.arn}:*"
+      },
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "task_execution_ecr_pull" {
+  role       = aws_iam_role.task_execution.name
+  policy_arn = module.ecr.policy_pull_arn
+}
+
+// EFS filesystem used for persistent storage across tasks. The filesystem is
+// created only if it's meant to be mounted inside the tasks.
+
+module "efs" {
+  for_each = var.mount_efs == null ? toset([]) : toset([var.name])
+  source   = "../efs-filesystem"
+
+  name          = "${var.env}--${var.name}"
+  allow_subnets = var.cluster_config.subnet_ids
+}
+
+resource "aws_iam_role_policy_attachment" "use_efs" {
+  for_each = var.mount_efs == null ? toset([]) : toset([var.name])
+
+  role       = aws_iam_role.task.name
+  policy_arn = module.efs[var.name].root_policy_arn
+}
+
+// Task definition of the application, which specifies which containers should
+// be part of the task and which settings do they have.
+
+data "aws_ssm_parameter" "task" {
+  for_each = toset(values(var.secrets))
+  name     = each.value
+}
+
+resource "aws_cloudwatch_log_group" "task" {
+  name              = "/${var.env}/${var.name}"
+  retention_in_days = 7
+}
+
+resource "aws_ecs_task_definition" "task" {
+  family       = var.name
+  cpu          = var.cpu
+  memory       = var.memory
+  network_mode = "awsvpc"
+
+  task_role_arn            = aws_iam_role.task.arn
+  execution_role_arn       = aws_iam_role.task_execution.arn
+  requires_compatibilities = ["FARGATE"]
+
+  container_definitions = jsonencode([
+    {
+      name      = "app"
+      image     = module.ecr.url
+      essential = true
+
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          "awslogs-group"         = aws_cloudwatch_log_group.task.name
+          "awslogs-region"        = "us-west-1"
```

Comment:
Should we change this to `us-east-2` based on our [decision re. AWS regions](https://forge.rust-lang.org/infra/docs/aws-regions.html)?